src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,190 @@
 
-        self.request = BasicParameter("{}")
-        self.action = BasicParameter("{}")
-
-        # daos_shell options
         self.hostlist = FormattedParameter("-l {}")
         self.hostfile = FormattedParameter("-f {}")
         self.configpath = FormattedParameter("-o {}")
         self.insecure = FormattedParameter("-i", None)
+        self.debug = FormattedParameter("-d", None)
+        self.json = FormattedParameter("-j", None)
+        self.subcommand_list = {
+            "format": self.format,
+            "prepare": self.prepare,
+        }
+
+    def __str__(self):
+        """Return the command with all of its defined parameters as a string.
+
+        Returns:
+            str: the command with all the defined parameters
+
+        """
+        # Join all the parameters that have been assigned a value with the
+        # command to create the command string
+        params = []
+        for name in self.get_param_names():
+            value = str(getattr(self, name))
+            if name == "action" and value in self.subcommand_list:
+                params.append(str(self.subcommand_list[value]))
+            elif value != "":
+                params.append(value)
+        return " ".join([os.path.join(self._path, self._command)] + params)
+
+    class DmgFormatSubCommand(CommandWithParameters):
+        """Defines a object representing a sub dmg (or daos_shell) command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgFormatSubCommand, self).__init__("format")
+            self.force = FormattedParameter("-f", None)
+
+    class DmgPrepareSubCommand(CommandWithParameters):
+        """Defines a object representing a sub dmg (or daos_shell) command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgPrepareSubCommand, self).__init__("prepare")
+            self.pci_wl = FormattedParameter("-w {}")
+            self.hugepages = FormattedParameter("-p {}")
+            self.targetuser = FormattedParameter("-u {}")
+            self.nvmeonly = FormattedParameter("-n", None)
+            self.scmonly = FormattedParameter("-s", None)
+            self.force = FormattedParameter("-f", None)
+            self.reset = FormattedParameter("--reset", None)
+
+
+def storage_scan(hosts):
+    """ Execute scan command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run scan on.
+        path (str, optional): Path to dmg command binary. Defaults to "".
+
+    Returns:
+        False if issue running command. True otherwise.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.request.value = "storage"
+    dmg.action.value = "scan"
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+
+    try:
+        result = dmg.run()
+    except process.CmdError as details:
```

Comment:
This particular exception is caught by `ExecutableCommand.run()` and in its place it will return a `CommandFailure` exception, which is what should be caught here.

Reply:
Ok. Will change.